t/Path-Class.t

```diff
@@ -0,0 +1,49 @@
+use strict;
+use warnings;
+use Test::More;
+
+eval "require Path::Class";
+plan skip_all => "Test::Pod 1.00 required for testing POD" if $@;
+
+use File::Path qw(remove_tree make_path);
+Path::Class->import;
+
+my $name = 'test';
+my $dir = dir($name);
+
+sub test {
+  my ($dir, $pass_arg) = @_;
+
+  my $args = [ $dir, ($pass_arg ? {} : ()) ];
+  my $desc = sprintf(
+    'dir isa %s, second arg is %s',
+    (ref($dir) || 'string'),
+    ($pass_arg ? '{}' : 'not passed')
+  );
+
+  return ($args, $desc);
+}
+
+for my $mk_dir ($name, dir($name)) {
+  for my $mk_pass_arg (0, 1) {
+
+    for my $rm_dir ($name, dir($name)) {
+      for my $rm_pass_arg (0, 1) {
+        remove_tree($name) if -e $name;
+
+        my ($mk_args, $mk_desc) = test($mk_dir, $mk_pass_arg);
+        make_path(@$mk_args);
+
+        if (ok( -d $dir, "we made $dir ($mk_desc)")) {
+          my ($rm_args, $rm_desc) = test($rm_dir, $rm_pass_arg);
+          remove_tree(@$rm_args);
+          ok( ! -d $dir, "...then we removed $dir ($rm_desc)");
+        } else {
+          fail("...can't remove it if we didn't create it");
+        }
+      }
+    }
+  }
+}
+
+done_testing;
```

Comment:
The use of Test::More::done_testing() will limit the backwards-compatibility of this distribution with respect to earlier versions of Perl 5.  According to the Changes file is the  Test-Simple distribution, done_testing() was introduced in Test-Simple version 0.87_01  Sun Mar 29 09:56:52 BST 2009.  So, per 'corelist -a Test::More', the earliest production version of Perl 5 that will be handle done_testing() without a problem will be 5.10.1.

I recommend counting the tests and declaring a plan.  Also, it would be a good idea to use 'perlbrew' or one of its allegedly superior replacements to locally install some earlier versions of Perl 5 and test the File-Path test suite against it.

Thank you very much.
Jim Keenan